### ISSUE TITLE
Live Preview Issues with Links, Bold, Italics, etc.

### ISSUE THREAD

**Issue by martin-wall** [tgrosinger/advanced-tables-obsidian#145]
Hey. I suspect you're aware of this already but I can't see any existing issues/comments regarding this, so making an issue just in case. You may also tell me this isn't changing!

**Describe the bug**

Following the live preview update, Advanced Tables is not adding the correct amount of spaces to other cells in the column that don't have the same formatting.

Take the following code.

```
| Column 1   | Column 2 | Column 3 |
| ---------- | -------- | -------- |
| **CELL1A** | CELL2A   | CELL3A   |
| CELL1B     | CELL2B   | CELL3B   |
```

This is displayed as follows:

![image](https://user-images.githubusercontent.com/8067056/147596928-ba586a6a-ba59-46e0-b6ed-4efb3cdbc3e3.png)

As we can see, the padding has not been added to CELL1A. Safe to assume that Advanced Tables is including the ** before and after the value in the padding, therefore presuming no padding is required. This happens with italics, links, etc.

**To Reproduce**

Use the following example:

```
| Column 1   | Column 2 | Column 3 |
| ---------- | -------- | -------- |
| **CELL1A** | CELL2A   | CELL3A   |
| CELL1B     | CELL2B   | CELL3B   |
```

**Expected behavior**

It would be good if the now hidden formatting is ignored by Advanced Tables and the correct padding is added.

**Screenshots**

![image](https://user-images.githubusercontent.com/8067056/147596928-ba586a6a-ba59-46e0-b6ed-4efb3cdbc3e3.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Obsidian v0.13.14
 - Advanced Tables Version: 0.15.0

**Comment by tgrosinger** [comment]
Hey, thanks for writing this up. This is a very good description of the problem. Trouble is, I am not sure what the best solution is.

If you take the file which is "incorrect" in Life Preview, and run the "Toggle Live Preview/Source mode" command, you'll see that it actually is formatted correctly in the raw text. It's just that when Live Preview renders it, the formatting characters are hidden.

Here's a demo:

![LivePreviewFormattingTable](https://user-images.githubusercontent.com/597386/147598729-5fad74a3-484d-4491-a825-89700dedeb9e.gif)

So far, the only solution I can think of would be to have tables fully rendered when in Live Preview, rather than the half render that is done now. However that is quite a big departure from how this plugin currently functions.

**Comment by martin-wall** [comment]
Thanks for the response. I won't be much use here as I am not much of a developer, so not a clue what is and isn't possible. Hopefully (or not?) someone else comes across this themselves and can provide some more valuable input than I am about to.

From my overly simplistic perspective, I thought you could have some sort of detection (or flag in the settings?) to state that you're using live preview mode, and if so, ignore markdown syntax such as **X** or [[X]] when calculating the number of spaces that need adding as padding.

**Comment by ashemedai** [comment]
I would add a vote to full rendering in Live Preview.

I wonder if watching `MarkdownViewModeType` is the way forward. The API change a while ago removed `live` and only left `source` and `preview`. Given that it can expand to source on-the-fly when you move the cursor over it, it would be weird if there was not some event trigger to hook into.

Mmm, looking further, their new dependencies explicitly list [CodeMirror](https://codemirror.net)'s state and view. I wonder if the live preview works in similar manner as the boolean toggle on [this decoration page](https://codemirror.net/6/examples/decoration/).

This is just speculation though, I have not played with Obsidian's API.

**Comment by Danathar** [comment]
Ran into this problem as well. It's a little annoying because I kinda like the live preview. I had to turn it off though because I have notes with tables using the plugins.

**Comment by onepeloton-kiva** [comment]
@Danathar I highly recommend setting up a hotkey to toggle between live preview and source mode while this is getting worked out, so you can use still use live preview in notes without tables (also because it's nice in general to be able to get back to source mode at a moment's notice for note refactoring)

**Comment by j-r-e-i-d** [comment]
Would love for this great plugin to work in Live Preview as it works in the legacy editor. Wish I had the skills to help.

**Comment by tgrosinger** [comment]
Just an update, there is rumor that Obsidian may natively support editing tables when in Live Preview. If/When that happens, I will update this plugin to only take effect when in Source Mode. This task is on hold until I hear more about plans from the Obsidian team.

**Comment by jgonggrijp** [comment]
@tgrosinger In case the Obsidian team does not go through with that plan: maybe you can take inspiration from the way the maintainers of the Admonition plugin have implemented it. They show the whole thing fully rendered when the cursor is elsewhere and the whole thing in source mode when the cursor is inside, exactly as you suggested, and exactly as I agree it should work.

**Comment by tgrosinger** [comment]
@jgonggrijp, thanks for the suggestions. Unfortunately that requires an amount of time which I will not be able to put into this plugin.

**Comment by Nixxen** [comment]
Looking at this, I think it might be the same issue I was having with latex formatting in issue #169 , but for other kinds of formatting syntax.

**Comment by ashemedai** [comment]
From the 0.13.20 changelog:

>There is now a new `editorLivePreviewField: StateField<boolean>` that can be used to check whether Live Preview is enabled in the editor.

**Comment by OnkelTem** [comment]
Just my few cents. I would also support the idea of all-or-nothing approach, i.e. showing the table source once a cursor is anywhere inside the table. However I'm not a plugin developer and I simply don't know how difficult or possible it is.

**Comment by aycajr** [comment]
> @tgrosinger In case the Obsidian team does not go through with that plan: maybe you can take inspiration from the way the maintainers of the Admonition plugin have implemented it. They show the whole thing fully rendered when the cursor is elsewhere and the whole thing in source mode when the cursor is inside, exactly as you suggested, and exactly as I agree it should work.

Hi guys, any plans to make 'Advanced tables' work properly within Admonition anytime soon?

**Comment by Elaws** [comment]
Hello, are there any news on that great feature request ?
Thanks !

**Comment by jasal82** [comment]
Table rendering has now been added in the latest update.

**Comment by jgonggrijp** [comment]
Nice! It seems to work exactly the way everyone wanted it to work. I think we can close this one, @tgrosinger .

**Comment by m0rtis0** [comment]
Is the initial issued resolved now? When I use the plugin in Obsidian with live view on it still looks like the first post

**Comment by rafaelurben** [comment]
> Is the initial issued resolved now? When I use the plugin in Obsidian with live view on it still looks like the first post

Yes, the issue is resolved. Have you updated Obsidian?

**Comment by jgonggrijp** [comment]
@m0rtis0 Do you see this when the text cursor is inside or outside the table?

**Comment by tgrosinger** [comment]
This could potentially be resolved by #199 if anyone wants to take an attempt at that.
